src/convertToObject.js

```diff
@@ -6,7 +6,55 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.length) {
+    return {};
+  }
+
+  const rawLines = sourceString.split(';');
+
+  // filter lines without letters
+  const noEmptyLines = rawLines
+    .map((line) => {
+      return line.trim();
+    })
+    .filter((item) => item.length > 0);
+
+  const aggregatedLines = noEmptyLines.reduce((prev, line) => {
+    if (line.includes(':')) {
+      prev.push(line);
+    } else {
+      prev[prev.length - 1] += line;
+    }
+
+    return prev;
+  }, []);
+
+  // remove ';'
+  const noSemicolon = aggregatedLines.map((item) => {
+    // Array[string]
+    const [property, value] = item.split(':');
+    const newValue = value.split(';').join('');
+
+    return `${property}:${newValue}`;
+  });
+
+  // trim
+  const noSpaces = noSemicolon.map((item) => {
+    const [property, value] = item.split(':');
+
+    const newKey = property.trim();
+    const newValue = value.trim();
+
+    return `${newKey}:${newValue}`;
+  });
```

Comment:
[CODE STYLE] - All chained methods called on the same level should have the same indentation. This makes code much more readable. You can chain this `map` directly after the previous operations.